### ISSUE TITLE
LineTouchData touchCallback firing with incorrect data and at wrong times

### ISSUE THREAD

**Issue by ToyWalrus** [imaNNeoFighT/fl_chart#276]
When using the library I keep experiencing a bug with the `touchCallback` in `LineTouchData`. When the callback is triggered, `LineTouchResponse` data is sent to the given callback, but the callback is called twice after a brief period if the user keeps the touch in same spot (holding their finger down; for example, I press the graph and the callback triggers with the data on the spot I pressed. If I keep holding my finger in the same spot, the following happens...)

After the initial call when the user presses down, the next call happens about half a second later and the `data.lineBarSpots` is empty, so the state values I'm setting get cleared (**Expected behavior**: this only gets called when the gesture ends, i.e. finger up) Then the callback is immediately called again with the previous values that the user still has selected (**Expected behavior** : this call shouldn't be happening, as the values have already been set and nothing has changed). When the user finishes the touch event (finger up), the callback isn't called again (**Expected behavior**: it should be called again with empty `data.lineBarSpots`, like the first call), thus leaving my state believing that the user is still holding their finger down, so it keeps displaying what should have been cleared.

```
  LineTouchData _createCallbacks(bool enabled) {
    final tooltipLineColor = Colors.gray[400];
    final touchThreshold = 12;
    return LineTouchData(
      enabled: enabled,
      fullHeightTouchLine: true,
      touchSpotThreshold: touchThreshold,
      touchCallback: (data) => _onSelectionChanged(data, enabled),
      touchTooltipData: LineTouchTooltipData(getTooltipItems: (items) => List.generate(items.length, (_) => null)),
      getTouchedSpotIndicator: (barData, indexes) => List<TouchedSpotIndicatorData>.generate(
        indexes.length,
        (_) => TouchedSpotIndicatorData(
            FlLine(color: tooltipLineColor, strokeWidth: .75),
            FlDotData(dotColor: barData.colors[0], dotSize: 4)
        )
      )
    );
  }

...

  void _onSelectionChanged(LineTouchResponse data, bool enabled) {
    if (!enabled) return;
    if (data.lineBarSpots.isEmpty) {
      _clearCurrentValues();
      return;
    }
    ... (set state values here)
  }
```

<img width="712" alt="Screen Shot 2020-03-31 at 11 53 02 AM" src="https://user-images.githubusercontent.com/25328178/78059108-913fbd80-7346-11ea-96db-ada4378b4ca5.png">

**Versions**
 - Flutter version: 1.12.13+hotfix.8
 - FlChart version: ^0.8.1

**Comment by imaNNeoFighT** [comment]
Hi,
thanks for the explanation,
we are going to update our touch handling flow,
please stay tuned, and check it with the new version.
Thanks!

**Comment by ToyWalrus** [comment]
Here's a video I recorded of the issue I described above.
Note that this is using fl_chart **version 0.9.0**

https://vimeo.com/408995691

**Comment by ToyWalrus** [comment]
After looking deeper into the code, I found the cause for this. The problem was that in my `touchCallback()` method, I was listening for the `(LineTouchResponse) data.lineBarSpots.isEmpty` to clear my data. However, looking at the trigger for the callback in `line_chart.dart:103`, the `FlLongPressEnd` is being passed with the location of where the press ended instead of `Offset.zero` like in the `onPanCancel/End` callback on line 127 and 139 of the same file. This in turn made the press end event essentially like another press/pan update event in that it would return the data points at that position, instead of no data points like if you were to use `Offset.zero` instead of `d.localPosition`.

I would have submitted a PR to fix this, but I haven't seen anybody else talk about this issue and it just seems to be an error in my own assumptions of how the touch callback worked. I fixed it in my local code by checking that `data.touchInput is FlLongPressEnd || data.lineBarSpots.isEmpty` instead of just the second clause.

So this behavior may or may not be WAD, but it is a little inconsistent that the `onPanCancel/End` callbacks will pass `Offset.zero` as their position but the `onLongPressEnd` callback doesn't.
